lib/router/src/utils.ts

```diff
@@ -47,7 +46,7 @@ export const parsePath: (path?: string) => StoryData = memoize(1000)(
 
     if (path) {
       const [, viewMode, storyId] = path.match(splitPathRegex) || [undefined, undefined, undefined];
-      if (viewMode && !viewMode.match(knownNonViewModesRegex)) {
+      if (viewMode) {
```

Comment:
I'm not 100% sure removing `knownNonViewModesRegex` It seems like any registered addons can be one of `viewMode` and for this PR `settings` also can be a `viewMode` so It is not needed anymore?  I referenced this [one](https://github.com/storybookjs/storybook/pull/6551/files#diff-18efc3f2c976feeab3887b2fd2a58f7e)